pkg/polaris/aws/aws.go

```diff
@@ -929,3 +929,148 @@ func (a API) TrustPolicies(ctx context.Context, id IdentityFunc, features []core
+		return fmt.Errorf("failed to get vpcs from RSC: %s", err)
+	}
+
+	vpcSyncedToRsc := slices.ContainsFunc(vpcs, func(vpc aws.AwsCloudAccountListVpcs) bool {
+		return vpc.VpcID == input.VmConfig.Vpc
+	})
+	if !vpcSyncedToRsc {
+		return fmt.Errorf("vpc %s does not exist in RSC AWS account %s for region %s. Check the VPC ID and region. If this was recently created, wait a few minutes and try again", input.VmConfig.Vpc, account.NativeID, input.Region)
+	}
+
+	// Validate Instance Profile exists in RSC metadata via AllAwsInstanceProfileNames
+	instanceProfiles, err := aws.Wrap(a.client).AllAwsInstanceProfileNames(ctx, account.ID, inputRegion)
+	if err != nil {
+		return fmt.Errorf("failed to get instance profiles: %s", err)
+	}
+	validInstanceProfile := slices.Contains(instanceProfiles, input.VmConfig.InstanceProfileName)
+	if !validInstanceProfile {
+		return fmt.Errorf("instance profile %s does not exist in RSC AWS account %s", input.VmConfig.InstanceProfileName, account.NativeID)
+	}
+
+	// Validate Subnet exists in RSC metadata via AwsCloudAccountListSubnets
+	subnets, err := aws.Wrap(a.client).AwsCloudAccountListSubnets(ctx, input.CloudAccountID, inputRegion, input.VmConfig.Vpc)
+	if err != nil {
+		return fmt.Errorf("failed to get subnets: %s", err)
+	}
+	validSubnet := slices.ContainsFunc(subnets, func(subnet aws.AwsCloudAccountSubnets) bool {
+		return subnet.SubnetID == input.VmConfig.Subnet
+	})
+	if !validSubnet {
+		return fmt.Errorf("subnet %s does not exist in RSC AWS account %s", input.VmConfig.Subnet, account.NativeID)
+	}
+
+	// Validate Security Groups
+	securityGroups, err := aws.Wrap(a.client).AwsCloudAccountListSecurityGroups(ctx, input.CloudAccountID, inputRegion, input.VmConfig.Vpc)
+	if err != nil {
+		return fmt.Errorf("failed to get security groups: %s", err)
+	}
+	// Validate Security Groups - check that all provided security groups exist
+	for _, inputSG := range input.VmConfig.SecurityGroups {
+		validSecurityGroup := slices.ContainsFunc(securityGroups, func(securityGroup aws.AwsCloudAccountSecurityGroups) bool {
+			return securityGroup.SecurityGroupID == inputSG
+		})
+		if !validSecurityGroup {
+			return fmt.Errorf("security group %s does not exist in RSC AWS account %s", inputSG, account.NativeID)
+		}
+	}
+
+	// Validate CloudCluster Request
+	ok, err := aws.Wrap(a.client).ValidateCreateAwsClusterInput(ctx, input)
+	if err != nil {
+		return fmt.Errorf("failed to validate create cloud cluster: %s", err)
+	}
+
+	if ok {
+		jobID, err := aws.Wrap(a.client).CreateAwsCloudCluster(ctx, input)
+		if err != nil {
+			return fmt.Errorf("failed to create cloud cluster: %s", err)
+		}
+		status, err := core.Wrap(a.client).WaitForTaskChain(ctx, jobID, 10*time.Second)
+		if err != nil {
+			return err
+		}
+		taskChain := status == core.TaskChainSucceeded
+		if !taskChain {
+			return fmt.Errorf("cloud cluster create task chain failed")
+		}
+	} else {
+		return fmt.Errorf("cloud cluster create validation failed")
```

Comment:
The GQL request seems to return both `isSuccessful` and `message`, do we want to propagate the message if validation fail?